src/animated/Controller.ts

```diff
@@ -163,21 +154,52 @@ class Controller<DS extends object = any> {
     return this
   }
 
-  // Called by the frameloop
-  onFrame(isActive: boolean) {
-    if (this.props.onFrame) {
-      this.props.onFrame(this.values)
+  destroy() {
+    this.stop()
+    this.props = {} as any
+    this.merged = {} as any
+    this.values = {} as any
+    this.interpolations = {}
+    this.animations = {}
+    this.configs = []
+  }
+
+  // Add this controller to the frameloop
+  private _start(onEnd?: OnEnd) {
+    if (onEnd) this.onEndQueue.push(onEnd)
+    if (this.idle) {
+      this.idle = false
+      start(this)
     }
-    if (!isActive) {
-      this._stop(true)
+  }
+
+  private _stop(finished?: boolean) {
+    this.idle = true
+    stop(this)
+    if (finished && this.props.onRest) {
+      this.props.onRest(this.merged)
+    }
+    if (this.onEndQueue.length) {
+      this.onEndQueue.forEach(onEnd => onEnd(finished))
+      this.onEndQueue = []
+    }
+  }
+
+  private _run(guid: number, props: UpdateProps<DS>, onEnd?: OnEnd) {
+    // Never call `onStart` for immediate animations
+    if (!props.immediate) {
```

Comment:
This fixes #362 by not calling `onStart` for immediate animations.

Reply:
I just realized that `immediate` can be a function, so the presence of `props.immediate` is not enough to know whether `onStart` should be called.

Reply:
oh, right, theres a helper for it, callProp, it's used throughout the lib